### ISSUE TITLE
[dotAI] : Index Build has problems handling Custom Fields

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#29233]
### Parent Issue

https://github.com/dotCMS/core/issues/28990

### Problem Statement

As a dotCMS User, I want to be able to build an Index inside the `dotAI` portlet that has a Custom Field. Currently, such field types are not getting picked up by the Embeddings API.

### Steps to Reproduce

* Set up the `dotAI` App, and include the `dotAI` portlet in the back-end.
* Create a simple VTL file with this path: `/application/vtl/custom-fields/my-file.vtl`
* Create a Content Type which has these fields:
  * `Title` - Text Field
  * `Preview` - Custom Field. Add this value:
```vtl
#dotParse("/application/vtl/custom-fields/my-file.vtl")
```
* Go to **dotAI > Manage Embeddings/Indexes**, and enter the following Lucene query, using your Content Type's var name:
```lucene
+contentType:{{your-content-type-var-name}}
```
* Click the `Build Index` button. Check the logs, and you'll see an error like this:
```
15:55:51.049  INFO  api.EmbeddingsAPIImpl - No valid fields to embed for:MyTesCT id:f4c9c9898e5bae39b363accb69c00a9b title:This is my title
```

### Acceptance Criteria

All field of data type `LONG_TEXT` should be parseable by the Embeddings API.

### dotCMS Version

24.04.24_LTS_V4, up to latest master

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

**Comment by jcastro-dotcms** [comment]
As per discussion with @john-thomas-dotcms and @dsilvam , we agreed on the expected behavior of how Custom Fields must be read when generating the embeddings. Here are the conclusions:

* I was able to confirm that the existing code already handles Custom Fields correctly.
* It's very important to note that the developer of the code in the Custom Field is responsible for passing down the final value to the HTML field representing the Velocity Variable Name that dotCMS is expecting. There are Custom Fields that simply display thumbnails for a given binary file in a Contentlet. So, if no value is specified for such a field, dotCMS won't have any data to save. If you need more information on how to accomplish this, please refer to the `URL Title` field in the `Blog` Content Type in our https://demo.dotcms.com/dotAdmin Site.
* When indexing fields, it's very important to note that Contentlets whose content from indexable fields is lower than the minimum character length WILL NOT be added to the index. You can update this value via the `com.dotcms.ai.embeddings.minimum.text.length` property, which defaults to `64` currently.

In summary, dotAI is handling Custom Fields as it should.

**Comment by victoralfaro-dotcms** [comment]
Docker image: `dotcms/dotcms:trunk_b00ffaf`
![image](https://github.com/user-attachments/assets/6fa9e06d-9993-4aac-a5fb-bae53ba6acfd)
![image](https://github.com/user-attachments/assets/f93b6412-a653-4907-9757-ef4314a71fdb)
![image](https://github.com/user-attachments/assets/361dca27-0519-4fe7-83b6-cc91339bb2b2)
